pulsar-client/src/main/java/org/apache/pulsar/client/impl/HttpClient.java

```diff
@@ -127,8 +128,9 @@ public void close() throws IOException {
     public <T> CompletableFuture<T> get(String path, Class<T> clazz) {
         final CompletableFuture<T> future = new CompletableFuture<>();
         try {
-            String requestUrl = new URL(serviceNameResolver.resolveHostUri().toURL(), path).toString();
-            String remoteHostName = serviceNameResolver.resolveHostUri().getHost();
```

Comment:
@sijie maybe a bug here, the host in requestUrl not equal with remoteHostName if multi-broker service url provided, and will step forward two service url every request. please take a look.

Reply:
sorry I don't quite get this. Can you show me an example?

Reply:
the PulsarServiceNameResolver.resolveHostUri() called twice(every time it will return the next address) inside HttpClient.get() , 
if service url is http://ip1:port1,ip2:port2, we assume requestUrl is ip1:port1, the  remoteHostName will be ip2, and the next time call HttpClient.get() the requestUrl is still ip1:port1, the remoteHostName is ip2. so HttpClient.get() will only request ip1:port1

Reply:
oh I see. yes. the original code has a bug there.